src/sidebar/services/api.js

```diff
@@ -226,30 +217,70 @@ export class APIService {
         onRequestFinished: store.apiRequestFinished,
       });
 
+    // Define available API calls.
+    //
+    // The type syntax is APICall<Parameters, Body, Result>, where `void` means
+    // no body / empty response.
+
+    /**
+     * @typedef AnnotationSearchResult
+     * @prop {Annotation[]} rows
+     * @prop {Annotation[]} replies
+     * @prop {number} total
+     */
+
+    /** @type {APICall<object, void, AnnotationSearchResult>} */
```

Comment:
The `object` type for the parameters of `/api/search` is just a placeholder, to be filled out more completely at a later time.